src/storage/checkpoint_io.cpp

```diff
@@ -0,0 +1,60 @@
+#include "storage/checkpoint_io.h"
+#include <vector>
+namespace terrier::storage {
+
+void BufferedTupleWriter::SerializeTuple(ProjectedColumns::RowView *row, ProjectedRow *row_buffer,
+                                         const storage::BlockLayout &layout) {
+  // First construct tuple in row_buffer, and calculate total size required
+  int32_t varlen_offset = 0;
+  std::vector<const VarlenEntry *> varlen_entries;
+  for (uint16_t projection_list_idx = 0; projection_list_idx < row->NumColumns(); projection_list_idx++) {
+    if (row->IsNull(projection_list_idx)) {
```

Comment:
`AccessForceNotNull` modifies the underlying data structure (it forces the bitmask to not null) and returns a pointer.  You should really be using `AccessWithNullCheck` for `row` and comparing it to `nullptr` here to condition on when the underlying data is null.

Reply:
Thanks for the advice. We changed our code in the latest version.